### PR TITLE
MEM Based DLC Unpacker

### LINKED ISSUE #76: DLC Unpacker for ME3
The DLC unpacker for ME3Explorer was removed by me a long time ago (probably 3.0.2). It has needed replacement with MEM code for some time.

As of the writing of this issue, the Unpacker has been implemented into the MEMDecompression branch. A few edge cases need to be tested, such as when an empty SFAR file is found but the mount.dlc file (used to detect unpack) is missing. Currently an exception is thrown, however it is unknown if this is how we should handle this and present it to the user.

Normally this is not an issue, however we immediately write a blank SFAR file once it has been loaded into memory. If the operation is canceled, the DLC is now invalid. Perhaps we can write the loaded SFAR back to disk in this instance, or ask the user what to do.

Other things:
 - Ensure there is enough disk space before starting the unpack thread

### PR DESCRIPTION
**Pull request by aquadran**
This PR contains a new DLC Unpacker based on MEM's DLC unpacking code. It is literally hundreds of times faster than the old code in ME3Explorer when extracting DLC.

Supports:
- Safe cancellation (will cleanup unpacked files if window is closed during unpack). SFAR is not rewritten unless the SFAR has fully unpacked
 - Size calculations
- Fancy progress bars
- Crazy fast unpack speed

Credits to aquadran for porting it over for us.

This pull request will close #76 when merged.

### CODE REVIEW

File: Licenses/COPYING_7ZIP.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+LZMA SDK is written and placed in the public domain by Igor Pavlov.
+
+You can copy, modify, distribute and perform LZMA SDK code, even for
```
Comment (Mgamerz) [inline Licenses/COPYING_7ZIP.txt:3]: Once merged we need to pull in the licenses for other projects into the licensing folder.

Comment (aquadran) [reply Licenses/COPYING_7ZIP.txt:3]: Yes, we need to

**Review by Mgamerz — APPROVED** [review]
